### PR TITLE
140 new data source securityroles

### LINKED ISSUE #139: [New Data Source]: /security/roles
### Description

/security/roles

### Requested Resource(s) and/or Data Source(s)

netapp-ontap_security_roles

### Potential Terraform Configuration

_No response_

### References

https://docs.netapp.com/us-en/ontap-restapi/ontap/swagger-ui/index.html#/security/role_collection_get

### Would you like to implement a fix?

None

### LINKED ISSUE #140: [New Resource]: /security/roles
### Description

/security/roles 

### Requested Resource(s) and/or Data Source(s)

* netapp-ontap_security_roles

### Potential Terraform Configuration

_No response_

### References

https://docs.netapp.com/us-en/ontap-restapi/ontap/swagger-ui/index.html#/security/role_create

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by wenjun666**

### CODE REVIEW

File: internal/provider/security/security_role_data_source.go [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/interfaces"
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/utils"
+)
+
+// TODO:
```
Comment (chuyich) [inline internal/provider/security/security_role_data_source.go]: Can you remove TODO?

File: internal/interfaces/security_role.go [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/mitchellh/mapstructure"
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/restclient"
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/utils"
+)
+
+// TODO:
```
Comment (chuyich) [inline internal/interfaces/security_role.go]: Same here: remove TODO

File: internal/interfaces/security_role.go [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+	SVMName string `mapstructure:"owner.name"`
+	Scope   string `mapstructure:"scope"`
+}
+
+// GetSecurityRoleByName to get security_role info
+func GetSecurityRoleByName(errorHandler *utils.ErrorHandler, r restclient.RestClient, name string, svmUUID string) (*SecurityRoleGetDataModelONTAP, error) {
+	api := "security/roles/" + svmUUID + "/" + name
+	query := r.NewQuery()
+	query.Set("name", name)
+	// if svmName == "" {
```
Comment (chuyich) [inline internal/interfaces/security_role.go]: What are these comments for? Can they be removed?

**Comment by chuyich** [comment]
For the data source, just wondered those "optional" parameters in the schema should be "computed". Can you double check?

**Review by suhasbshekar — CHANGES_REQUESTED** [review]
CHANGELOG?

File: docs/data-sources/security_role_data_source.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  Retrieves a Security role
+---
+
+# netapp-ontap_security_role_data_source (Data Source)
+
+SecurityRole data source
+
+## Example Usage
+```terraform
+data "netapp-ontap_security_role_data_source" "security_role" {
```
Comment (carchi8py) [inline docs/data-sources/security_role_data_source.md]: Remove data_source

File: docs/data-sources/security_role_data_source.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "netapp-ontap_security_role_data_source Data Source - terraform-provider-netapp-ontap"
```
Comment (carchi8py) [inline docs/data-sources/security_role_data_source.md]: data_source

File: docs/data-sources/security_roles_data_source.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "netapp-ontap_security_roles_data_source Data Source - terraform-provider-netapp-ontap"
```
Comment (carchi8py) [inline docs/data-sources/security_roles_data_source.md]: data_source

File: docs/data-sources/security_roles_data_source.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+description: |-
+  Retrieves Security Rules
+---
+
+# netapp-ontap_security_roles_data_source (Data Source)
+
+SecurityRules data source
+
+## Example Usage
+data "netapp-ontap_security_roles_data_source" "security_roles" {
```
Comment (carchi8py) [inline docs/data-sources/security_roles_data_source.md]: data_source

**Review by carchi8py — CHANGES_REQUESTED** [review]
Remove data source from the names completely in all places.

**Comment by chuyich** [comment]
Missing the examples

**Comment by carchi8py** [comment]
What do we need to get this merged in?

File: CHANGELOG.md [hunk 8]
```diff
@@ -2,6 +2,7 @@
 
 FEATURES:
 * **New Data Source:** `netapp-ontap_volumes_files` ([#8](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/8))
+* **New Data Source:** `netapp-ontap_security_role` ([#139](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/139))
```
Comment (chuyich) [inline CHANGELOG.md:7]: Missing netapp-ontap_security_roles. here.

**Review by chuyich — CHANGES_REQUESTED** [review]

**Comment by wenjun666** [comment]
> For the data source, just wondered those "optional" parameters in the schema should be "computed". Can you double check?

Yes, should be "computed."

File: internal/provider/security/security_roles_data_source.go [hunk 9]
```diff
@@ -0,0 +1,231 @@
...
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/interfaces"
+	"github.com/netapp/terraform-provider-netapp-ontap/internal/utils"
+)
+
+// TODO:
+// copy this file to match you data source (should match internal/provider/security_role_data_source.go)
+// replace SecurityRules with the name of the resource, following go conventions, eg IPInterfaces
+// replace security_roles with the name of the resource, for logging purposes, eg ip_interfaces
+// make sure to create internal/interfaces/security_role.go too)
+// delete these 5 lines
```
Comment (suhasbshekar) [inline internal/provider/security/security_roles_data_source.go]: can you remove the TODO?

**Review by suhasbshekar — CHANGES_REQUESTED** [review]
Looking good, just a TODO to be removed

**Review by carchi8py — CHANGES_REQUESTED** [review]
Can you make the change @suhasbshekar requested.

**Review by carchi8py — APPROVED** [review]

**Review by chuyich — APPROVED** [review]

**Review by suhasbshekar — APPROVED** [review]

**Pull request merged**